sycl/include/CL/sycl/detail/kernel_program_cache.hpp

```diff
@@ -25,29 +25,42 @@ namespace detail {
 class context_impl;
 class KernelProgramCache {
 public:
-  /// Denotes pointer to some entity with its state.
+  /// Denotes build error data. The data is filled in from cl::sycl::exception
+  /// class instance.
+  struct BuildError {
+    std::string Msg;
+    pi_int32 Code;
+
+    /// Equals to true if the Msg and Code are initialized. This flag is added
+    /// due to the possibility of error code being equal to zero even in case 
+    /// if build is failed and cl::sycl::exception is thrown.
+    bool FilledIn;
```

Comment:
Isn't `Msg.empty()` == `FilledIn`?

Reply:
I believe it may be. Though, flag is more explicit data and more understandable to read. On the other hand the struct may have a distinct method `bool isFilledIn() const`.

Reply:
+1 for replacing this flag with public member function.
`FilledIn` seems to duplicate information available in two other fields and maintaining is not necessary.